libsweep/src/sweep.cc

```diff
@@ -399,6 +641,51 @@ void sweep_device_set_sample_rate(sweep_device_s device, int32_t hz, sweep_error
     sweep::protocol::error_destruct(protocolerror);
     return;
   }
+
+  // Check the status bytes do not indicate failure
+  const uint8_t status_bytes[2] = {response.cmdStatusByte1, response.cmdStatusByte2};
+  int32_t status_code = sweep::protocol::ascii_bytes_to_integral(status_bytes);
+  switch (status_code) {
+  case 11:
```

Comment:
11 is a failure status code that indicates an invalid parameter. These details are outlined in the protocol docs, but I think it might be best to include them in the protocol implementation. Perhaps making a container of possible status codes for each type of response?